packages/common/src/audius-query/hooks/usePaginatedQuery.ts

```diff
@@ -60,57 +60,65 @@ export const useAllPaginatedQuery = <
   const [page, setPage] = useState(0)
   const [status, setStatus] = useState<Status>(Status.IDLE)
   const [allData, setAllData] = useState<Data[]>([])
-  const args = {
-    ...baseArgs,
-    limit: pageSize,
-    offset: page * pageSize
-  } as ArgsType
-  const result = useQueryHook(args, queryHookOptions)
 
   useCustomCompareEffect(
     () => {
       setAllData([])
+      setPage(0)
       setLoadingMore(false)
     },
     [baseArgs],
     isEqual
   )
 
+  const args = {
+    ...baseArgs,
+    limit: pageSize,
+    offset: page * pageSize
+  } as ArgsType
+  const result = useQueryHook(args, queryHookOptions)
+
   useCustomCompareEffect(
     () => {
-      if (!statusIsNotFinalized(result.status)) {
+      setStatus(result.status)
+      if (result.status === Status.ERROR) {
+        setLoadingMore(false)
+        return
+      }
+      if (result.status === Status.SUCCESS) {
+        setAllData((allData) => [...allData, ...result.data])
         setLoadingMore(false)
       }
-      setStatus(result.status)
-      if (result.status !== Status.SUCCESS) return
-
-      setAllData((allData) => [...allData, ...result.data])
     },
     [result.status, args],
     isEqual
   )
 
   const notError = result.status !== Status.ERROR
+  const notStillLoadingCurrentPage =
+    !loadingMore && !(result.status === Status.LOADING)
   const notStarted = result.status === Status.IDLE && allData.length === 0
   const hasNotFetched = !result.data && result.status !== Status.SUCCESS
   const fetchedFullPreviousPage = result.data?.length === pageSize
 
   const hasMore =
-    notError && (notStarted || hasNotFetched || fetchedFullPreviousPage)
+    notError &&
+    notStillLoadingCurrentPage &&
+    (notStarted || hasNotFetched || fetchedFullPreviousPage)
 
   const loadMore = useCallback(() => {
-    if (loadingMore) {
+    if (!notStillLoadingCurrentPage) {
```

Comment:
if we're going to negate this here, I think I'd rather have the var be `isStillLoadingCurrentPage` and use `!` when we want the negative case. Avoids the double negative

Reply:
would it be better to check `hasMore` here since it includes the loadingCurrentPage check?

Reply:
i prefer the loading more since it's clearer but i think hasmore would probably work too!

Reply:
I'm happy either way